src/shared/src/backend-supports/project-supports.ts

```diff
@@ -90,11 +91,29 @@ const calculateProjectStartDate = (wps: { duration: number; startDate: Date }[])
   return minDate;
 };
 
+// calculate the project's status based on its pacakges' status
```

Comment:
can you explain the logic of the function using js doc pls (if you do like /** then press enter it will open the whole thing like i had in my refactoring PR)